File: coredata/ObjC/_BloodPressureWrapper.m
#import "_BloodPressureWrapper.h"
#import "BloodPressure.h"

@interface _BloodPressureWrapper ()
@end

@implementation _BloodPressureWrapper {
}
- (BloodPressure *)entity {
    return (BloodPressure *) [super entity];
}

- (instancetype)initWithEntity:(BloodPressure *)entity {
    self = [super initWithEntity:entity];
    if (self) {
        self.bloodPressure = entity.bloodPressure;
    }
    return self;
}

- (void)updateEntity:(BloodPressure *)entity {
    [super updateEntity:entity];
    entity.bloodPressure = self.bloodPressure;
}
@end

File: coredata/Swift/_BloodPressureWrapper.swift
import Foundation

class _BloodPressureWrapper: HealthDataWrapper {
    override func entity() -> BloodPressure? {
        return _entity as BloodPressure?
    }
    var bloodPressure: NSNumber?

    init(bloodPressure: BloodPressure?) {
        super.init(healthData: bloodPressure)
        if let e = bloodPressure {
            self.bloodPressure = e.bloodPressure
        }
    }
    func updateBloodPressure(bloodPressure: BloodPressure) {
        super.updateHealthData(bloodPressure)
        bloodPressure.bloodPressure = self.bloodPressure
    }
}
